### PR TITLE
feat(kno-4306): add objects bulk endpoints support

### PR DESCRIPTION
**Pull request by bceskavich**
- Adds support for the new bulk add subscriptions API
- Backfills missing support for the existing bulk set + delete objects APIs

### CODE REVIEW

File: knock/objects.go [hunk 1]
```diff
@@ -211,6 +230,17 @@ type AddSubscriptionsResponse struct {
 	Subscriptions []*ObjectSubscription
 }
 
+type BulkAddObjectSubscription struct {
+	ID         string                 `json:"id"`
+	Properties map[string]interface{} `json:"properties,omitempty"`
+	Recipients []interface{}          `json:"recipients"`
```
Comment (brentjanderson) [inline knock/objects.go:236]: Not my expertise, and I see we use this same type elsewhere, but why don't we have a dedicated type for Recipients beyond `interface{}`?

File: knock/objects.go [hunk 2]
```diff
@@ -356,6 +390,28 @@ func (os *objectsService) Set(ctx context.Context, setObjectRequest *SetObjectRe
...
+	}
+
+	bulkSetObjectsRes := BulkRequestResponse{BulkOperation: &BulkOperation{}}
+
+	_, err = os.client.do(ctx, req, bulkSetObjectsRes.BulkOperation)
+	if err != nil {
+		return nil, errors.Wrap(err, "error making request to bulk set objects")
+	}
+
+	if err != nil {
```
Comment (brentjanderson) [inline knock/objects.go:408]: Can this be called if we already checked it on line 404?

File: knock/objects.go [hunk 3]
```diff
@@ -372,6 +428,28 @@ func (os *objectsService) Delete(ctx context.Context, deleteObjectRequest *Delet
...
+	}
+
+	bulkDeleteObjectsRes := BulkRequestResponse{BulkOperation: &BulkOperation{}}
+
+	_, err = os.client.do(ctx, req, bulkDeleteObjectsRes.BulkOperation)
+	if err != nil {
+		return nil, errors.Wrap(err, "error making request to bulk delete objects")
+	}
+
+	if err != nil {
```
Comment (brentjanderson) [inline knock/objects.go:446]: Same q here as above about error checking.

File: knock/objects.go [hunk 4]
```diff
@@ -548,6 +626,28 @@ func (os *objectsService) AddSubscriptions(ctx context.Context, addSubscriptions
...
+	}
+
+	bulkAddSubscriptionsRes := BulkRequestResponse{BulkOperation: &BulkOperation{}}
+
+	_, err = os.client.do(ctx, req, bulkAddSubscriptionsRes.BulkOperation)
+	if err != nil {
+		return nil, errors.Wrap(err, "error making request to bulk add subscriptions")
+	}
+
+	if err != nil {
```
Comment (brentjanderson) [inline knock/objects.go:644]: And here.

**Review by brentjanderson — APPROVED** [review]
Not going to block this, but I have some questions about a couple of error checks that seem to be redundant.

File: knock/objects.go (hunk 1 above)
Comment (bceskavich) [reply knock/objects.go:236]: I don't think we have a good reason, other than consistency with existing patterns.

File: knock/objects.go (hunk 2 above)
Comment (bceskavich) [reply knock/objects.go:408]: Ah yup it's duplicate, thanks for catching!

**Pull request merged**